### PR TITLE
Stateful updates

### PR DESCRIPTION
**Pull request by pzbitskiy**
* New enum values
* New opcodes
* Change hashing ops cost

### CODE REVIEW

File: dev/TEAL_opcodes.md [hunk 1]
```diff
@@ -24,23 +24,23 @@ Ops have a 'cost' of 1 unless otherwise specified.
 - Pops: *... stack*, []byte
 - Pushes: []byte
 - SHA256 hash of value X, yields [32]byte
-- **Cost**: 7
+- **Cost**: 35
```
Comment (algobolson) [inline dev/TEAL_opcodes.md]: We need a way to say that at version 1 it was 7 and at version 2 it is 35 (and also make that true in Check/Eval)

File: dev/TEAL_opcodes.md [hunk 2]
```diff
@@ -414,6 +416,7 @@ FirstValidTime causes the program to fail. The field is reserved for future use.
 | 3 | ZeroAddress | []byte | 32 byte address of all zero bytes |
 | 4 | GroupSize | uint64 | Number of transactions in this atomic transaction group. At least 1. |
 | 5 | LogicSigVersion | uint64 |  |
+| 6 | Round | uint64 | Current round number |
```
Comment (algobolson) [inline dev/TEAL_opcodes.md:425]: can we bring back real time while we're at it? If not the current-block according to the proposer, then the time of the previous block. Just for stateful-eval.

File: dev/TEAL_opcodes.md [hunk 3]
```diff
@@ -554,9 +557,20 @@ See `bnz` for details on how branches work. `b` always jumps to the offset.
 
 params: account index, application id (top of the stack on opcode entry)
 
-## app_local_get
+## app_local_gets
 
 - Opcode: 0x62
+- Pops: *... stack*, {uint64 A}, {[]byte B}
+- Pushes: any
+- read from account's A from local state of the current application key B  => value
```
Comment (algobolson) [inline dev/TEAL_opcodes.md]: maybe
"read from account specified by Txn.Account[A] ..."

**Review by algobolson — CHANGES_REQUESTED** [review]
Need the per-version cost.
A nice-to-have feature request and idea about re-wording a doc.

File: dev/TEAL_opcodes.md (hunk 1 above)
Comment (pzbitskiy) [reply dev/TEAL_opcodes.md]: done

File: dev/TEAL_opcodes.md (hunk 3 above)
Comment (pzbitskiy) [reply dev/TEAL_opcodes.md]: done for this and all other

**Pull request merged**